community/rfcs/MM-RAG-RFG.md

```diff
@@ -0,0 +1,173 @@
+
+MM-RAG is critical for OPEA since it will:
+1. define new interfaces for OPEA's microservices to work with multimodal data. Such interfaces will not only extend langchain interfaces but also make use of (if-existing) interfaces for multimodal data in popular frameworks such as Huggingface, OpenAI, etc...
+2. introduce new microservices that can handle multimodal data.
+3. enable new AI applications that enterprises need.
+
+## Design Proposal
+This RFC proposes a Multimodal Retrieval-Augmented Generation (MM-RAG) architecture detailed in the following figure. 
+
+![MM-RAG Architecture](./assets/mm-rag-architecture.png)
+
+The proposed architecture involves the creation of two megaservices. 
+- The first megaservice functions as the core pipeline, comprising four microservices: embedding, retriever, reranking, and LVLM. This megaservice exposes a MMRagBasedVisualQnAGateway, allowing users to query the system via the `/v1/mmrag_visual_qna` endpoint.
+- The second megaservice manages user data storage in VectorStore and is composed of a single microservice, embedding. This megaservice provides a MMRagDataIngestionGateway, enabling user access through the `/v1/mmrag_data_ingestion` endpoint.
+- The third megaservice functions as a helper to extract list of frame-transcript pairs from videos using audio-to-text models (e.g., BLIP2) for transcripting or LVLM model (e.g., LLAVA) for captioning. This megaservice is composed of 2 microservices: transcripting and LVLM. This megaservice provides a MMRagVideoprepGateway, enabling user access through the `/v1/mmrag_video_prep` endpoint.
+- **Optional** The fourth megaservice functions as a helper to extract list of image-caption pairs from documents (e.g., research papers & journals). This megaservice provides a MMRagDocprepGateway, enabling user access through the `/v1/mmrag_doc_prep` endpoint.
+
+
+For this architecture and in order to guarantee the reusability and expansion ability of this architecture for other applications in future, we are going to propose the followings.
+
+### 1. Data Classes
+
+MM-RAG introduces new data classes. In `comps/cores/proto/docarray`:
+- `ImageDoc(BaseDoc)` (cf., docarray.documents.ImageDoc) , 
+- `VideoDoc(BaseDoc)` (cf., docarray.documents.VideoDoc),
+- `AudioDoc(BaseDoc)` (cf., docarray.documents.AudioDoc), 
+- `TextAndImageDoc(BaseDoc)` contains information for an image-text pair
+- `MultimodalDoc = Union[TextDoc, ImageDoc, TextImageDoc]`
+- `SearchedMultimodalDoc(MultimodalDoc)` which is similar to SearchedDoc but for multimodal data
+
+### 2. Interfaces
+MM-RAG defines new interfaces used in OPEA's microservices:
+#### 2.1 Embeddings
+- Interface `MultimodalEmbeddings` that extends the interface langchain_core.embeddings.Embeddings with an abstract method: 
+```python 
+embed_multimodal_document(self, doc: MultimodalDoc) -> List[float]

```

Comment:
The overall RFC is good to me. just minor comment here. I think this interface is implementation specific one, right?. my point here is you don't need to mention that. you just need tell user what's the standard input and output just like you did in Data Classes section, that's enough.